### ISSUE TITLE
Bug: [new config system] `ignores` does not work without `files` 

### ISSUE THREAD

**Issue by filetvignon** [eslint/eslint#17103]
### Environment

Node version: 18.16.0
npm version: 9.6.4
Local ESLint version: 8.38.0
Global ESLint version: N/A
Operating System: Ubuntu 22.04 (WSL)

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```js
[
  {
    rules: {
      "no-unused-vars": "error",
    },
  },
  {
    ignores: ["error.js"],
    rules: {
      "no-unused-vars": "warn",
    },
  },
];
```

</details>

`./error.js` and `./warn.js`:

```js
let unusedVar;
```

Then run `eslint .`

### What did you expect to happen?

```shell
{cwd}/error.js
  1:5  error  'unusedVar' is defined but never used  no-unused-vars

{cwd}/warn.js
  1:5  warning  'unusedVar' is defined but never used  no-unused-vars
```

### What actually happened?

```shell
{cwd}/error.js
  1:5  warning  'unusedVar' is defined but never used  no-unused-vars

{cwd}/warn.js
  1:5  warning  'unusedVar' is defined but never used  no-unused-vars
```

### Link to Minimal Reproducible Example

https://github.com/filetvignon/eslint-bug-ignores

### Participation

- [X] I am willing to submit a pull request for this issue.

### Additional comments

I set 2 configuration objects: one with a rule all files and another to change that rule for all files EXCEPT `error.js`.
I omitted the `files` property in these objects; although the documentation recommends using `ignores` with `files`, [it states that this should work](https://eslint.org/docs/latest/use/configure/configuration-files-new#excluding-files-with-ignores) and omitting `files` should be the same as using `files: '**/*'`.

In reality, it works only if I explicitly set `files` in the second configuration object:

```js
[
  {
    rules: {
      "no-unused-vars": "error",
    },
  },
  {
    files: ["**/*"],
    ignores: ["error.js"],
    rules: {
      "no-unused-vars": "warn",
    },
  },
];
```

**Comment by fasttime** [comment]
I am able to reproduce this. This is happening because configs without `files` are considered [universal configs](https://github.com/humanwhocodes/config-array/blob/8e17f4a7378cb0b417e1103d60ef397b26d2f917/src/config-array.js#L707-L711), and they are applied to all files regardless of `ignores`. I also find this behavior surprising, but I believe that it's there intentionally to allow excluding things like `node_modules` from linting. In this case we should update the documentation to mention that `ignores` works differently in a config without `files`.

Maybe @nzakas can advise the best course of action here.

**Comment by fasttime** [comment]
Thanks for the clarification @mdjermanovic. Sometimes it's hard to know if it's the implementation that doesn't match the docs or vice-versa. That said, I find the headline of the example mentioned above confusing, as it should actually read (bold is my correction):

> If `ignores` is used without `files` **but with some other settings**, then the configuration object applies to all files except the ones specified in `ignores`, for example: 

As for the bug, it needs to be fixed in [`@humanwhocodes/config-array`](https://github.com/humanwhocodes/config-array), although it would be also a good idea to add some unit tests here in `eslint`.

@filetvignon would you like to submit a pull request?

**Comment by mdjermanovic** [comment]
> That said, I find the headline of the example mentioned above confusing, as it should actually read (bold is my correction):
> 
> > If `ignores` is used without `files` **but with some other settings**, then the configuration object applies to all files except the ones specified in `ignores`, for example

Yes, I think we should fix that in the docs.

**Comment by nzakas** [comment]
Just catching up on this now. This does look like a bug, and the docs could be clearer, I'd suggest something more along the lines of this:

> If `ignores` is used without `files` and there are other keys (such as `rules`), then the configuration object applies to all files except the ones specified in ignores, for example

**Comment by nzakas** [comment]
I pushed a new version of @humanwhocodes/config-array, but I won't have time to integrate into ESLint before traveling for next week.

**Comment by mdjermanovic** [comment]
> I pushed a new version of @humanwhocodes/config-array, but I won't have time to integrate into ESLint before traveling for next week.

I can take this, but it seems that the new version 0.11.9 hasn't been released. The release workflow failed on `npm ci`:

https://github.com/humanwhocodes/config-array/actions/runs/4961978783/jobs/8879447925

**Comment by nzakas** [comment]
Sorry about that, should be published now.
